applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -211,15 +211,15 @@ def ImportModelPart(self):
             self._ImportModelPart(self.main_model_part, self.settings["model_import_settings"])
 
     def PrepareModelPart(self):
+        # This will be removed once the Model is fully supported! => It wont e necessary anymore
+        if not self.model.HasModelPart(self.main_model_part.Name):
```

Comment:
I dont think this can be here
The CheckAndPrepare things are executed afterwards, this means that the submodelparts that are created in there wont be in the first lvl of the model, which is currently still needed!

Why did you move it there?

Reply:
In that case should be moved before assigning the materials